dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventCategoryOptValidationHookTest.java

```diff
@@ -187,7 +190,9 @@ void testDefaultCocWithNonDefaultCatCombo()
 
         // then
         assertTrue( reporter.hasErrors() );
-        assertThat( reporter.getReportList().get( 0 ).getErrorCode(), is( TrackerErrorCode.E1055 ) );
+        assertTrue( reporter.hasErrorReport( err -> E1055.equals( err.getErrorCode() ) &&
+            EVENT.equals( err.getTrackerType() ) &&
+            event.getUid().equals( err.getUid() ) ) );
```

Comment:
I just picked this example for no particular reason - there are plenty others in this PR - when using `assertTrue`/`assertFalse` I highly recommend to create a helper method `assertHasError` or something. With such a method we can then justify to do multiple checks or provide lengthy messages because this code is just occurring once. Also there should at some point be a better way of asserting than `assertTrue`/`assertFalse`  because something has been added to the API we test we can switch using the new method in one place.